### PR TITLE
Ftr datepicker stencil one (Closes #39)

### LINKED ISSUE #39: Calendar component
**Is your feature request related to a problem? Please describe.**
Why isn't calendar built in as a native input type?! Also, I haven't seen any date range pickers.

**Describe the solution you'd like**
I want a clean, extensible, responsive web-component calendar. It could be used to select single dates or date ranges. It could also be used to simply display information, but maybe that's getting too far into specific use-cases.

**Describe alternatives you've considered**
There are too many to name!

**Additional context**

Screen Shot 2019-05-07 at 12 28 34 PM

Screen Shot 2019-05-07 at 12 27 22 PM

### PR DESCRIPTION
**Pull request by tikri**
# Pull Request

## Type of change

- [x] New feature (a non-breaking change)

## Description

Implemented datepicker component

Fixes #39

## Checklist

- [x Local build is still running with my changes
- [x] I added tests 
- [x] New and existing unit test pass locally with my changes
- [x] My changes do not contain any dead or commented out code
- [x] I added documentation for public functionality
- [x] I added comments, at least in hard-to-understand areas

### CODE REVIEW

File: rollup-plugins/helpers/moment-locales-to-modules.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
+const glob = require('glob');
+const fs = require('fs');
+const moment = require('moment');
+
+const regex = 'function\\s\\(moment\\)\\s{([.\\S\\s]*)}';
+// const regex = '(week\\s:\\s{[.\\S\\s]*?})|(((weekdays|months|week)\\w*\\s):.*)';
```
Comment (Cr45hCode) [inline rollup-plugins/helpers/moment-locales-to-modules.js]: delete
```suggestion
```

File: rollup-plugins/orgenic-ui-post-processing.js [hunk 2]
```diff
@@ -13,12 +14,13 @@ function orgenicUiPostProcessing(config) {
...
 
         generateBundle: function (options, bundle) {
-            if (config.target !== 'start' && !options.amd.id) {
+            if (config.target !== 'start' && options.entryFileNames.indexOf('.esm.js') < 0) {
                 return;
             }
 
             return new Promise(resolve => {
                 const delay = config.target === 'build' ? 4000 : 2000;
+                // momentLocales2Modules({ outDir: 'www/orgenic-ui-assets/og-calendar-locales' });
```
Comment (Cr45hCode) [inline rollup-plugins/orgenic-ui-post-processing.js]: delete outcommented code
```suggestion
```

File: src/components/og-calendar/og-calendar.e2e.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
+import { newE2EPage, E2EElement, E2EPage } from '@stencil/core/testing';
+import moment from '../og-internal-calendar/interfaces/node_modules/moment';
+import { isNull } from 'util';
+
+describe('og-calendar', () => {
+    let page: E2EPage;
+    let component: E2EElement;
+    let flyout: E2EElement;
+
+    // enum Month
```
Comment (Cr45hCode) [inline src/components/og-calendar/og-calendar.e2e.ts]: delete outcommented code
```suggestion
```

File: src/components/og-datepicker/og-datepicker.e2e.ts [hunk 4]
```diff
@@ -0,0 +1,71 @@
+import { newE2EPage, E2EElement, E2EPage } from '@stencil/core/testing';
+
+describe('og-datepicker', () => {
+    let page: E2EPage;
+    let component: E2EElement;
+    let flyout: E2EElement;
+    let header: E2EElement;
+    let input: E2EElement;
+
+    // enum Month
```
Comment (Cr45hCode) [inline src/components/og-datepicker/og-datepicker.e2e.ts]: delete outcommented code
```suggestion
```

File: src/components/og-datepicker/og-datepicker.e2e.ts [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+        expect(flyout).not.toHaveClass(
+            'og-datepicker__flyout__calendar--visible'
+        );
+    });
+
+    it('displays the date in the right format', async () => {
+        expect(await input.getProperty('value')).toEqualText('30-05-2019');
+    });
+
+    // todo: check this
```
Comment (Cr45hCode) [inline src/components/og-datepicker/og-datepicker.e2e.ts]: did someone had a look at this?

File: src/components/og-internal-calendar/og-internal-calendar.scss [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+    text-transform: uppercase;
+    // font-size: 12px;
+
+}
+
+.day {
+    background: var(--og-calendar__day-Background);
+    color: var(--og-calendar__day-Color);
+    text-align: center;
+    // font-size: 12px;
```
Comment (Cr45hCode) [inline src/components/og-internal-calendar/og-internal-calendar.scss]: delete
```suggestion
```

File: src/components/og-internal-calendar/og-internal-calendar.scss [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+}
+
+td {
+    line-height: 1;
+}
+
+th {
+    text-align: center;
+    text-transform: uppercase;
+    // font-size: 12px;
```
Comment (Cr45hCode) [inline src/components/og-internal-calendar/og-internal-calendar.scss]: delete
```suggestion
```

File: src/components/og-internal-calendar/og-internal-calendar.scss [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+    --og-calendar__day-Background: var(--og-calendar__day-Background--active, none);
+    --og-calendar__day-Color: var(--og-calendar__day-Color--active);
+}
+
+
+
+.week {
+    padding-right: 8px;
+    text-align: right;
+    // font-size: 12px;
```
Comment (Cr45hCode) [inline src/components/og-internal-calendar/og-internal-calendar.scss]: delete
```suggestion
```

File: src/components/og-datepicker/og-datepicker.e2e.ts (hunk 5 above)
Comment (tikri) [reply src/components/og-datepicker/og-datepicker.e2e.ts]: https://github.com/ionic-team/stencil/issues/1530

**Pull request merged**